apps/prairielearn/assets/scripts/lib/questionsTable.ts

```diff
@@ -0,0 +1,368 @@
+import { decodeData, onDocumentReady, parseHTMLElement } from '@prairielearn/browser-utils';
+import {
+  Tabulator,
+  FormatModule,
+  EditModule,
+  FilterModule,
+  SortModule,
+  PageModule,
+  MutatorModule,
+  FrozenColumnsModule,
+  CellComponent,
+  Editor,
+} from 'tabulator-tables';
+import { html } from '@prairielearn/html';
+import { uniq } from 'lodash';
+
+import type {
+  Topic,
+  Tag,
+  SharingSet,
+  AssessmentsFormatForQuestion,
+} from '../../../src/lib/db-types';
+import type { QuestionsPageDataAnsified } from '../../../src/models/questions';
+import { idsEqual } from '../../../src/lib/id';
+
+// Copied from QuestionTable.html.ts component. Not importing because the import process creates dependencies of files that are not compatible with tsc.
```

Comment:
@nwalters512 can you help me to figure out a way to declare this interface only once? If I declare it in src/components/QuestionsTable.html.ts and use it here, then some of the indirect imports in that component will cause an issue because they are still .js files (e.g., middlewares/staticNodeModules.js), and tsc doesn't seem to like that. Should I create a .d.ts file? Or is there a better way?

Reply:
Thanks, done in 5c23bbce8a35b10b0b906e61f65c862670e05776.